prov/efa/src/efa_ep.c

```diff
@@ -115,6 +118,24 @@ static int efa_ep_modify_qp_rst2rts(struct efa_qp *qp)
 				      IBV_QP_STATE | IBV_QP_SQ_PSN);
 }
 
+static int efa_ep_modify_qp_rst2rts_with_rnr_retry_modify(struct efa_qp *qp)
```

Comment:
I still don't understand why another function is needed for this. Can you change `efa_ep_modify_qp_rst2rts()` to set `attr.rnr_retry = 0;` and `IBV_QP_RNR_RETRY` when rdma-core has the appropriate symbols?

Also, `IBV_QP_RNR_RETRY` should not be set when the endpoint type is DGRAM. It's not going to hurt anything, but we don't want customers to think that DGRAM will do retries for them if they're looking at this code.